src/StreamJsonRpc/JsonRpcEventSource.cs

```diff
@@ -0,0 +1,232 @@
+        /// Cancel Request start.
+        /// </summary>
+        /// <param name="message">Cancellation message.</param>
+        [Event(CancelRequestStartEvent, Level = EventLevel.Verbose)]
+        public void CancelRequestStart(string message)
+        {
+            this.WriteEvent(CancelRequestStartEvent, message);
+        }
+
+        /// <summary>
+        /// Cancel Request end.
+        /// </summary>
+        /// <param name="message">Cancellation message.</param>
+        [Event(CancelRequestStopEvent, Level = EventLevel.Verbose)]
+        public void CancelRequestStop(string message)
+        {
+            this.WriteEvent(CancelRequestStopEvent, message);
+        }
+
+        /// <summary>
+        /// Invoke Notification.
+        /// </summary>
+        /// <param name="message">Details of the Notification.</param>
+        [Event(InvokeNotificationStartEvent, Level = EventLevel.Verbose)]
+        public void InvokeNotificationStart(string message)
+        {
+            this.WriteEvent(InvokeNotificationStartEvent, message);
+        }
+
+        /// <summary>
+        /// Notification completed.
+        /// </summary>
+        /// <param name="message">Details of the Notification.</param>
+        [Event(InvokeNotificationStopEvent, Level = EventLevel.Verbose)]
+        public void InvokeNotificationStop(string message)
+        {
+            this.WriteEvent(InvokeNotificationStopEvent, message);
+        }
+
+        /// <summary>
+        /// Creates a string representation of arguments of max 1024 characters for logging.
+        /// </summary>
+        /// <param name="arguments">A single argument or an array of arguments.</param>
+        /// <returns>String representation of first argument only.</returns>
+        internal static string GetArgumentsString(object arguments)
+        {
+            if (arguments == null)
+            {
+                return string.Empty;
+            }
+
+            const int maxLength = 1024;
+
+            if (arguments is object[] args)
+            {
+                var stringBuilder = new StringBuilder();
+                if (args != null && args.Length > 0)
+                {
+                    for (int i = 0; i < args.Length; ++i)
+                    {
+                        stringBuilder.Append($"arg{i}: {args[i].ToString()}, ");
+                        if (stringBuilder.Length > maxLength)
+                        {
+                            return $"{stringBuilder.ToString().Substring(0, maxLength)}...(truncated)";
+                        }
+                    }
+                }
+
+                return stringBuilder.ToString();
+            }
+            else
+            {
+                string retVal = arguments.ToString();
+                return (retVal.Length > maxLength) ? $"{retVal.Substring(0, maxLength)}...(truncated)" : retVal;
+            }
+        }
+
+        internal static string GetErrorDetails(JsonRpcError error)
```

Comment:
similar: unless we have a need for error details, I'd rather they not be included in ETW tracing.
Ideally the ETW traces should only include a fixed amount of data, or known to be very small (e.g. request id and method name is fine for requests, and id for all responses + error code for errors).

Reply:
Ok. I will remove error message. It will only have Id and error code.